vllm/config/vllm.py

```diff
@@ -248,81 +246,43 @@ def compute_hash(self) -> str:
-            vllm_factors.append("None")
-        if self.speculative_config:
-            vllm_factors.append(self.speculative_config.compute_hash())
-        else:
-            vllm_factors.append("None")
-        if self.structured_outputs_config:
-            vllm_factors.append(self.structured_outputs_config.compute_hash())
-        else:
-            vllm_factors.append("None")
-        vllm_factors.append(self.observability_config.compute_hash())
-        if self.quant_config:
-            pass  # should be captured by model_config.quantization
-        if self.compilation_config:
-            vllm_factors.append(self.compilation_config.compute_hash())
-        else:
-            vllm_factors.append("None")
-        if self.kv_transfer_config:
-            vllm_factors.append(self.kv_transfer_config.compute_hash())
-        else:
-            vllm_factors.append("None")
-        if self.ec_transfer_config:
-            vllm_factors.append(self.ec_transfer_config.compute_hash())
-        else:
-            vllm_factors.append("None")
+        def _compile(config_obj: SupportsCompileFactors | None) -> CompileFactors:
+            if config_obj is None:
+                return {}
+            return config_obj.compile_factors()
+
+        factors: dict[str, Any] = {
+            "version": __version__,
+            "model": _compile(self.model_config),
+            "cache": _compile(self.cache_config),
+            "parallel": _compile(self.parallel_config),
+            "scheduler": _compile(self.scheduler_config),
+            "device": _compile(self.device_config),
+            "load": _compile(self.load_config),
+            "speculative": _compile(self.speculative_config),
+            "structured_outputs": _compile(self.structured_outputs_config),
+            "observability": self.observability_config.compile_factors(),
+            "compilation": _compile(self.compilation_config),
+            "kv_transfer": _compile(self.kv_transfer_config),
+            "ec_transfer": _compile(self.ec_transfer_config),
+            "lora": _compile(self.lora_config),
+        }
+
+        factors["max_num_batched_tokens"] = self.scheduler_config.max_num_batched_tokens
+
         if self.additional_config:
-            if isinstance(additional_config := self.additional_config, dict):
-                additional_config_hash = safe_hash(
-                    json.dumps(additional_config, sort_keys=True).encode(),
-                    usedforsecurity=False,
-                ).hexdigest()
+            additional_config = self.additional_config
+            if isinstance(additional_config, dict):
+                factors["additional"] = additional_config
             else:
-                additional_config_hash = additional_config.compute_hash()
-            vllm_factors.append(additional_config_hash)
+                assert isinstance(additional_config, SupportsCompileFactors)
+                factors["additional"] = additional_config.compile_factors()
```

Comment:
Bare asserts in configs cause unhelpful Pydantic validation errors when they're raised

```py
            elif isinstance(additional_config, SupportsCompileFactors):
                factors["additional"] = additional_config.compile_factors()
            else:
                raise ValueError("...")
```